pkg/ottl/parser.go

```diff
@@ -56,12 +56,40 @@ func (s *Statement[K]) Execute(ctx context.Context, tCtx K) (any, bool, error) {
 	return result, condition, nil
 }
 
-func NewParser[K any](functions map[string]interface{}, pathParser PathExpressionParser[K], enumParser EnumParser, telemetrySettings component.TelemetrySettings) Parser[K] {
+func NewParser[K any](
+	functions map[string]interface{},
+	pathParser PathExpressionParser[K],
+	options ...Option,
+) Parser[K] {
+	var cfg config
+	for _, opt := range options {
+		opt(&cfg)
+	}
+
 	return Parser[K]{
 		functions:         functions,
 		pathParser:        pathParser,
-		enumParser:        enumParser,
-		telemetrySettings: telemetrySettings,
+		enumParser:        cfg.enumParser,
+		telemetrySettings: cfg.telemetrySettings,
+	}
+}
+
+type config struct {
+	enumParser        EnumParser
+	telemetrySettings component.TelemetrySettings
+}
+
+type Option func(*config)
+
+func WithEnumParser(parser EnumParser) Option {
+	return func(c *config) {
+		c.enumParser = parser
+	}
+}
+
+func WithTelemetrySettings(settings component.TelemetrySettings) Option {
```

Comment:
To play devil's advocate, should `TelemetrySettings` be required? It is available to all Collector components so it should be easily available to any OTTL host component. We haven't really used it yet, but I could see potential benefits arising from making stronger guarantees about the presence of `TelemetrySettings` in OTTL functions.

Reply:
Based on the conversations happening in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14712 I could see TelemetrySettings becoming required eventually.